lib/components/normal-components/Board.ts

```diff
@@ -558,25 +559,28 @@ export class Board
       return
     }
 
-    for (const child of this.children) {
-      if (child instanceof NormalComponent) {
-        let childOldCenter: { x: number; y: number } | undefined
+    if (this.subcircuit_id) {
+      const descendantIds = getDescendantSubcircuitIds(db, this.subcircuit_id)
+      const allowedSubcircuitIds = new Set([
+        this.subcircuit_id,
+        ...descendantIds,
+      ])
 
-        if (child.pcb_component_id) {
-          const comp = db.pcb_component.get(child.pcb_component_id)
-          if (comp) childOldCenter = comp.center
-        } else if (child instanceof Group && child.pcb_group_id) {
-          const group = db.pcb_group.get(child.pcb_group_id)
-          if (group) childOldCenter = group.center
+      const pcbAndCadElementsToMove = db.toArray().filter((elm) => {
+        if (elm.type === "pcb_board") return false
+        if (elm.type !== "cad_component" && !elm.type.startsWith("pcb_")) {
+          return false
         }
-
-        if (childOldCenter) {
-          child._repositionOnPcb({
-            x: childOldCenter.x + deltaX,
-            y: childOldCenter.y + deltaY,
-          })
+        if (
+          !("subcircuit_id" in elm) ||
+          typeof elm.subcircuit_id !== "string"
+        ) {
+          return false
```

Comment:
Lets not change this function with special cases, they arent obvious, all the changes here seem like a hack- why?